### PR TITLE
Finish merging protocol mapping and pipeline interface

### LINKED ISSUE #55: Proposal: merge pipeline interface and protocol mapping into one config file
To prevent relying on a hard-coded folder structure, let's merge protocol mappings into a component of pipeline_interface.yaml and move those up one level, and then make the paths to pipelines in protocol_mappings relative to this file; 

this eliminates the hard-coded paths of 'config' and 'pipelines' subfolders we are currently employing.

### LINKED ISSUE #88: increase verbosity of required inputs not found
Small thing: if a pipeline is not submitted because input files are not found, right now looper says:
```
 Not submitted: Required input files not found
```
I think in this case it should state specifically which files are missing. It does in `--debug` mode but this might be worth displaying to the user.

### PR DESCRIPTION
**Pull request by nsheff**
* Fixes #88
* Move toward #96 
* Fix #55

### CODE REVIEW

File: looper/looper.py [hunk 1]
```diff
@@ -251,25 +251,20 @@ def run(prj, args, remaining_args, interface_manager):
         # Go through all pipelines to submit for this protocol.
         # Note: control flow doesn't reach this point if variable "pipelines"
         # cannot be assigned (library/protocol missing).
-        for pipeline_interface, pipeline_job in pipelines:
+        for pipeline_interface, pipeline_key, pipeline_job in pipelines:
```
Comment (vreuter) [inline looper/looper.py]: nice value-agnostic name

File: looper/models.py [hunk 2]
```diff
@@ -2081,12 +2089,47 @@ def __init__(self, pipedir):
                 _LOGGER.error(str(iface))
                 raise e
 
+    def pipeline_key_to_path(self, pipeline_key):
```
Comment (vreuter) [inline looper/models.py]: It looks like `script_path` and `script_cmd` are swapped here?

**Comment by vreuter** [comment]
Looks great, I just think there may be a variable name swap in one of the functions.

File: looper/models.py [hunk 3]
```diff
@@ -2091,18 +2091,21 @@ def __init__(self, pipedir):
...
-        :param str pipeline_key: the key in the pipeline interface yaml file used
+        :param str pipeline_key: the key in the pipeline interface file used
             for the protocol_mappings section. Previously was the script name.
+        :return (str, str): more restrictive version of input key, along with
+            absolute path for pipeline script.
 
         """
-        # key may contain extra command-line flags; split out to strict key / flags
-        strict_pipeline_key, tmp, pipeline_key_args = pipeline_key.partition(' ')
+        # key may contain extra command-line flags; discard flags.
```
Comment (nsheff) [inline looper/models.py]: we don't discard flags; we retain them because we use them later

File: looper/models.py (hunk 2 above)
Comment (nsheff) [inline looper/models.py]: where?

File: looper/models.py [hunk 4]
```diff
@@ -1,28 +1,26 @@
...
 
 .. code-block:: python
 
     from models import Project
     prj = Project("config.yaml")
-    prj.add_sample_sheet()
     # that's it!
 
-Explore!
+Explore:
```
Comment (nsheff) [inline looper/models.py]: I got tired of it.

File: looper/looper.py (hunk 1 above)
Comment (nsheff) [inline looper/looper.py]: great!

File: looper/models.py (hunk 3 above)
Comment (vreuter) [inline looper/models.py]: I mean from the full command, to derive the key. But yes, it's too strong.

**Comment by nsheff** [comment]
I don't think they were swapped, but I renamed them so they are more clear now, is that better?

File: looper/models.py [hunk 5]
```diff
@@ -2018,16 +2023,11 @@ def build_pipelines(self, protocol_name, priority=True):
...
-                             for path in script_paths])
-
-        if priority and len(jobs) > 1:
-            return jobs[0]
+                jobs.append([(ifproto.interface, ) +
+                             ifproto.pipeline_key_to_path(pipeline_key)
+                             for pipeline_key in pipeline_keys])
 
-        return list(itertools.chain(*jobs))
+        return jobs[0] if priority and len(jobs) > 1 else itertools.chain(*jobs)
```
Comment (nsheff) [inline looper/models.py]: ?
```
Traceback (most recent call last):
  File "/home/ns5bc/.local/bin/looper", line 9, in <module>
    load_entry_point('looper==0.6.0.dev0', 'console_scripts', 'looper')()
  File "/home/ns5bc/.local/lib/python2.7/site-packages/looper/looper.py", line 816, in main
    run(prj, args, remaining_args, interface_manager=interface_manager)
  File "/home/ns5bc/.local/lib/python2.7/site-packages/looper/looper.py", line 235, in run
    if len(pipelines) == 0:
TypeError: object of type 'itertools.chain' has no len()
```

**Comment by vreuter** [comment]
Should also address https://github.com/epigen/looper/pull/122

**Pull request merged**